java/client/src/main/java/glide/api/commands/servermodules/GlideJson.java

```diff
@@ -0,0 +1,411 @@
+import glide.api.BaseClient;
+import glide.api.GlideClient;
+import glide.api.GlideClusterClient;
+import glide.api.models.ClusterValue;
+import glide.api.models.GlideString;
+import glide.api.models.commands.ConditionalChange;
+import glide.api.models.commands.json.JsonGetOptions;
+import glide.utils.ArgsBuilder;
+import glide.utils.ArrayTransformUtils;
+import java.util.concurrent.CompletableFuture;
+import lombok.NonNull;
+
+public class GlideJson {
+
+    public static final String JSON_PREFIX = "JSON.";
+    public static final String JSON_SET = JSON_PREFIX + "SET";
+    public static final String JSON_GET = JSON_PREFIX + "GET";
+
+    private GlideJson() {}
+
+    /**
+     * Sets the JSON value at the specified <code>path</code> stored at <code>key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param path Represents the path within the JSON document where the value will be set. The key
+     *     will be modified only if <code>value</code> is added as the last child in the specified
+     *     <code>path</code>, or if the specified <code>path</code> acts as the parent of a new child
+     *     being added.
+     * @param value The value to set at the specific path, in JSON formatted string.
+     * @return A simple <code>"OK"</code> response if the value is successfully set. If value isn't
+     *     set because of <code>set_condition</code>, returns null.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, "doc", , ".", "{'a': 1.0, 'b': 2}").get();
+     * assert value.equals("OK");
+     * }</pre>
+     */
+    public static CompletableFuture<String> set(
+            @NonNull BaseClient client,
+            @NonNull String key,
+            @NonNull String path,
+            @NonNull String value) {
+        return executeCommand(client, new String[] {JSON_SET, key, path, value})
+                .thenApply(r -> (String) r);
+    }
+
+    /**
+     * Sets the JSON value at the specified <code>path</code> stored at <code>key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param path Represents the path within the JSON document where the value will be set. The key
+     *     will be modified only if <code>value</code> is added as the last child in the specified
+     *     <code>path</code>, or if the specified <code>path</code> acts as the parent of a new child
+     *     being added.
+     * @param value The value to set at the specific path, in JSON formatted GlideString.
+     * @return A simple <code>"OK"</code> response if the value is successfully set. If value isn't
+     *     set because of <code>set_condition</code>, returns null.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, gs("doc"), , gs("."), gs("{'a': 1.0, 'b': 2}")).get();
```

Comment:
```suggestion
     * String value = GlideJson.set(client, gs("doc"), , gs("."), gs("{'a': 1.0, 'b': 2}")).get();
```

Reply:
or `Json.set` or `JSON.set`? We can rename the class.

Reply:
I'm on board with renaming the class `Json`, as for any Java class, it is always enforced to be in some package (namespace), so name conflicting shouldn't be a thing.